### PR TITLE
refactor: remove renderDashCardActions in favor of a component

### PR DESCRIPTION
**Pull request by npretto**
### Description

This refactors `renderDashCardActions` to a proper component, creating a `DashCardActionsPanel` component that can store the state isDashCardTabMenuOpen introduced in https://github.com/metabase/metabase/pull/35324, instead of leaving it in the DashCard component that doesn't care about it.

The diff of the PR can be confusing but I tried to make small changes in isolated commits:

- 06d277e312ca023f99506aac1fc07b340779e461 moves `DashboardCardActionsPanel` inside the `DashCardActionButtons` component as it's basically just a styled div.
- d5abde45f1713761e9a9c8b744e85add813fb3ee ^ allows moving isDashCardTabMenuOpen state inside of `DashCardActionButtons`
- 1167fb63375bceb38d68c073ead5649740a6f091 removes renderDashCardActions as it just renders a component now, I did some testing (with console logs) and I didn't see extra rerenders caused by removing `useCallback`
- e0d9ef004c5b4ef24ed829d6f755833f143aa3ff renames `DashCardActionButtons` to `DashCardActionsPanel` as now that component contains the whole panel and I think it's a more appropriate name for it

### CODE REVIEW

**Comment by npretto** [comment]
Current dependencies on/for this PR:
* master
  * **PR #35240** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35240" target="_blank">Graphite</a> 
    * **PR #35366** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35366" target="_blank">Graphite</a>  👈
    * **PR #35324** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35324" target="_blank">Graphite</a> 
      * **PR #35327** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35327" target="_blank">Graphite</a> 
    * **PR #35309** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35309" target="_blank">Graphite</a> 
      * **PR #35310** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35310" target="_blank">Graphite</a> 
    * **PR #35243** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35243" target="_blank">Graphite</a> 
    * **PR #35241** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35241" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/metabase/metabase/35366?utm_source=stack-comment).

File: frontend/src/metabase/dashboard/components/DashCard/DashCardActionButtons/DashCardActionsPanel.tsx [hunk 1]
```diff
@@ -40,11 +44,11 @@ interface Props {
   ) => void;
   showClickBehaviorSidebar: () => void;
   onPreviewToggle: () => void;
-  onDashCardMenuClose: () => void;
-  onDashCardMenuOpen: () => void;
+  onLeftEdge: boolean;
+  onMouseDown: (event: React.MouseEvent) => void;
```
Comment (kamilmielnik) [inline frontend/src/metabase/dashboard/components/DashCard/DashCardActionButtons/DashCardActionsPanel.tsx:48]: Hmm, `React` is not imported.
Also, we use React things without `React.` prefix more often.

File: frontend/src/metabase/dashboard/components/DashCard/DashCardActionButtons/index.ts [hunk 2]
```diff
@@ -1,2 +1 @@
-// eslint-disable-next-line import/no-default-export -- deprecated usage
-export { default } from "./DashCardActionButtons";
+export { DashCardActionsPanel } from "./DashCardActionsPanel";
```
Comment (kamilmielnik) [inline frontend/src/metabase/dashboard/components/DashCard/DashCardActionButtons/index.ts:1]: > - 06d277e312ca023f99506aac1fc07b340779e461 moves `DashboardCardActionsPanel` inside the `DashCardActionButtons` component as it's basically just a styled div.
> - e0d9ef004c5b4ef24ed829d6f755833f143aa3ff renames `DashCardActionButtons` to `DashCardActionsPanel` as now that component contains the whole panel and I think it's a more appropriate name for it

Now the directory is called `DashCardActionButtons` but the the only component it's exporting is `DashCardActionsPanel` (which does not mention `Buttons` in its name).

I think - despite props passing - we were better off with `DashboardCardActionsPanel` definition living in `DashCard`.

Comment (npretto) [reply frontend/src/metabase/dashboard/components/DashCard/DashCardActionButtons/index.ts:1]: Should I close the PR or is it just a matter of finding a better name?

Comment (kamilmielnik) [reply frontend/src/metabase/dashboard/components/DashCard/DashCardActionButtons/index.ts:1]: > Should I close the PR 

No, the PR gets rid of `renderDashCardActions` which is a good thing.

> is it just a matter of finding a better name?

A better name is one way to handle it.
Another is to keep the `isDashCardTabMenuOpen` state and `DashboardCardActionsPanel` definition in `DashCard`.
Also we could keep it as is, it's definitely not a blocking comment.

File: package.json [hunk 3]
```diff
@@ -13,10 +13,10 @@
     "@dnd-kit/core": "^6.0.8",
     "@dnd-kit/modifiers": "^6.0.1",
     "@dnd-kit/sortable": "^7.0.2",
-    "@emotion/cache": "^11.7.1",
+    "@emotion/cache": "^11.10.5",
```
Comment (kamilmielnik) [inline package.json:16]: I'd approve the PR but I see some unexpected changes (looks like target branch was updated?).

Comment (npretto) [reply package.json]: Thanks for point that out, I updated a stack with graphite that contained the parent of this PR

**Review by kamilmielnik — APPROVED** [review]
:+1: 

Approving but please think about [this comment](https://github.com/metabase/metabase/pull/35366#discussion_r1384487151) before merging.

File: frontend/src/metabase/dashboard/components/DashCard/DashCardActionButtons/index.ts (hunk 2 above)
Comment (npretto) [reply frontend/src/metabase/dashboard/components/DashCard/DashCardActionButtons/index.ts]: I restructured the files a bit, this is the resulting DashCard folder
image
The rationale is that `DashCardActionsPanel` uses all the buttons, that are now children of the `DashCardActionsPanel` folder (which is pretty much the same as renaming the `DashCardActionButtons` folder).
I also created some a folder for some of the buttons that were in a single file, in this way it should be clear that DashCardActionsPanel.tsx is the main component.

Do you think this looks better?

Comment (kamilmielnik) [reply frontend/src/metabase/dashboard/components/DashCard/DashCardActionButtons/index.ts]: Oh, nice! :+1:

**Pull request merged**